### PR TITLE
Minor fixes for dynamic reconfigure on R2000

### PR DESCRIPTION
**Pull request by bk-mtg**
- Fix conflict between dynamic reconfigure servers between R2000 and R2300
- Adjust some of the config enums for R2000 which were breaking the dynamic reconfigure GUI due to mismatches between enum values and defaults.

### CODE REVIEW

File: pf_driver/cfg/PFDriverR2000.cfg [hunk 1]
```diff
@@ -18,8 +18,8 @@ gen_r2000.add("subnet_mask",          str_t,    3, "IP netmask", "255.0.0.0")
 gen_r2000.add("gateway",              str_t,    4, "IP gateway", "0.0.0.0")
 gen_r2000.add("scan_frequency",       int_t,    5, "The parameter scan_frequency defines the set point for the rotational speed of the sensor head and therefore the number of scans recorded per second. For the R2000 valid values range from 10 Hz to 50 Hz with steps of 1 Hz.", 35, 10, 50)
 
-scan_direction_enum = gen_r2000.enum([ gen_r2000.const("cw",  int_t, 0, "Clocl-wise"),
-                                       gen_r2000.const("ccw", int_t, 1, "Counter clock-wise")],
+scan_direction_enum = gen_r2000.enum([ gen_r2000.const("cw",  str_t, "cw", "Clocl-wise"),
```
Comment (ipa-hsd) [inline pf_driver/cfg/PFDriverR2000.cfg:21]: "Clock-wise"

File: pf_driver/include/pf_driver/pf/pf_interface.h [hunk 2]
```diff
@@ -30,8 +30,16 @@ class PFInterface
         }
         protocol_interface_ = std::make_shared<PFSDPBase>(ip_);
 
-        param_server_R2000_.setCallback(boost::bind(&PFInterface::reconfig_callback_r2000, this, boost::placeholders::_1, boost::placeholders::_2));
-        param_server_R2300_.setCallback(boost::bind(&PFInterface::reconfig_callback_r2300, this, boost::placeholders::_1, boost::placeholders::_2));
+        if(expected_device_ == "R2000")
```
Comment (ipa-hsd) [inline pf_driver/include/pf_driver/pf/pf_interface.h:33]: I like this idea, but maybe it more sense to do this here https://github.com/PepperlFuchs/ROS_driver/blob/64e3620341323e6fac48a3259e5de55feeb60a87/pf_driver/src/pf/pf_interface.cpp#L62-L67

**Comment by ipa-hsd** [comment]
Overall it looks good. Added some comments for some minor changes.

File: pf_driver/cfg/PFDriverR2000.cfg (hunk 1 above)
Comment (bk-mtg) [reply pf_driver/cfg/PFDriverR2000.cfg:21]: Good catch! I'll add this when I do the other change.

File: pf_driver/include/pf_driver/pf/pf_interface.h (hunk 2 above)
Comment (bk-mtg) [reply pf_driver/include/pf_driver/pf/pf_interface.h:33]: I think that generally makes sense, though I'm not 100% sure I'd expect a "handle_version" function do be setting up reconfigure callbacks.  I think I might make a new function call to put in `init()` right after the call to `handle_version`...

**Comment by bk-mtg** [comment]
Made some updates per your suggestions.  I've verified that it compiles with the new changes, but I no longer have R2000 hardware to test with.

File: pf_driver/src/pf/pf_interface.cpp [hunk 3]
```diff
@@ -69,6 +70,20 @@ bool PFInterface::handle_version(int major_version, int minor_version)
     return false;
 }
 
+void PFInterface::setup_param_server()
+{
+    if(expected_device_ == "R2000")
```
Comment (ipa-hsd) [inline pf_driver/src/pf/pf_interface.cpp:75]: Sorry I am just nitpicking, but the variable name should be `product_`. But both are the same.

**Comment by ipa-hsd** [comment]
Even though I have added a minor comment, I will approve this.

**Pull request merged**